pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -190,15 +195,38 @@ func (a *AmbientIndexImpl) All() []*model.AddressInfo {
 	a.mu.RLock()
 	defer a.mu.RUnlock()
 	res := make([]*model.AddressInfo, 0, len(a.byUID)+len(a.serviceByNamespacedHostname))
+
+	addedWEAddr := map[networkAddress]*model.AddressInfo{}
+	allNetworkAddresses := sets.New[networkAddress]()
+
 	for _, wl := range a.byUID {
-		res = append(res, workloadToAddressInfo(wl.Workload))
+		addressInfo := workloadToAddressInfo(wl.Workload)
+		networkAddr := networkAddressFromWorkload(wl)
+		// For WorkloadEntry we need to check whether a workload with similar network/address exists.
+		// WorkloadEntry may have a single address.
+		if strings.Contains(wl.GetUid(), workloadEntryGroupKind) && len(networkAddr) > 0 {
+			if allNetworkAddresses.Contains(networkAddr[0]) {
+				// Workload with network/address similar to the WorkloadEntry network/address already exists.
+				// Skipping this WorkloadEntry.
+				log.Warnf("Skipping WorkloadEntry %s as it has the same address of another workload on the same network", wl.GetUid())
```

Comment:
can we note that this is only for ambient? just for extra context

Reply:
```go
log.Warnf("Skipping WorkloadEntry %s as in Ambient it can't have the same address of another workload on the same network", wl.GetUid())
```
Looks fine?